tonic/src/transport/channel/mod.rs

```diff
@@ -104,17 +106,26 @@ impl Channel {
     /// This creates a [`Channel`] that will load balance accross all the
     /// provided endpoints.
     pub fn balance_list(list: impl Iterator<Item = Endpoint>) -> Self {
-        let list = list.collect::<Vec<_>>();
-
-        let buffer_size = list
-            .iter()
-            .next()
-            .and_then(|e| e.buffer_size)
-            .unwrap_or(DEFAULT_BUFFER_SIZE);
+        let (channel, mut tx) = Self::balance_channel(DEFAULT_BUFFER_SIZE);
+        list.for_each(|endpoint| {
+            let _res = tx.try_send(Change::Insert(endpoint.uri.clone(), endpoint));
```

Comment:
_probably_ should just unwrap here if there is an error we want to know because that would mean there is a bug in this code!

Reply:
done